### ISSUE TITLE
Error on translation when adding a new theme

### ISSUE THREAD

**Issue by sarahdib** [PrestaShop/PrestaShop#23344]
This issue is related to the [Refactoring of Translation storage component](https://github.com/PrestaShop/PrestaShop/issues/19242) project

You can read about the project [here](https://github.com/PrestaShop/PrestaShop/issues/19242).

#### Describe the bug

After uploading a new theme and click on save there is an error 

#### Steps to Reproduce

1. Go to Bo -> Theme & logo
2. Click on add a new theme
3. Upload new theme and save

**Screenshots**

<img width="1259" alt="Capture d’écran 2021-02-22 à 12 18 15" src="https://user-images.githubusercontent.com/3034677/108702749-f75aa380-7509-11eb-88b6-8bc0cae6fa78.png">

#### Additional information

* PrestaShop version: develop
* PHP version: 7.1

**Comment by matks** [comment]
Will be fixed by https://github.com/PrestaShop/PrestaShop/pull/23127/files

**Comment by Progi1984** [comment]
Fixed by #23127

**Comment by khouloudbelguith** [comment]
Hi @sarahdib, @PrestaShop/prestashop-core-developers,

This issue is reproduced again as reported by @Robin-Fischer-PS.
In fact, this issue is fixed previously with this PR: 
https://github.com/PrestaShop/PrestaShop/pull/23127
https://drive.google.com/file/d/1KexDcEZs0GUYpRDIGEwu-adg7eDAaKJr/view?usp=sharing

But after merging this PR: https://github.com/PrestaShop/PrestaShop/pull/23296
The issue is here again
I checked with these steps:
```

1. git checkout 77992ca18d
2. composer install
3. Try to add a theme => see error

```
Thank you!
